R/format_PEW_temp.R

```diff
@@ -0,0 +1,541 @@
+      ReleaseAlive = CaptureAlive,
+      CapturePopID = PopID,
+      CapturePlot = NA_character_,
+      ReleasePopID = ifelse(ReleaseAlive == TRUE, CapturePopID, NA_character_),
+      ReleasePlot = ifelse(ReleaseAlive == TRUE, CapturePlot, NA_character_),
+      #### ASSUMPTION: use NA, as there is no information
+      OriginalTarsusMethod = NA_character_,
+      WingLength = NA,
+      ExperimentID = dplyr::case_when(Experiment == "BSM - Griffioen et al. 2019 PeerJ" ~
+                                        "COHORT; PARENTAGE",
+                                      Experiment == "2h Temp D4" ~ "SURVIVAL",
+                                      Experiment == "2h Temp D4 + Handicaping Male: Griffioen et al. 2019 Front Ecol&Evol" ~
+                                        "SURVIVAL; PARENTAGE"),
+      #### NOT SURE ABOUT THIS
+      Age_observed = case_when(Age == "0" & Method == "ChickRinging" ~ 1,
+                               Age == "1" ~ 5,
+                               Age == ">=2" ~ 7,
+                               #### Account for controls in December
+                               Age == "1" & Method == "Night Control Winter" &
+                                 lubridate::month(CaptureDate) == 12 ~ 6,
+                               #### Account for controls in January & February
+                               Age == ">=2" & Method == "Night Control Winter" &
+                                 lubridate::month(CaptureDate) %in% c(1, 2) ~ 8,
+                               is.na(Age) ~ 5),
+      Age_observed = as.integer(Age_observed))
+
+
+  Capture_data <-
+    Capture_data_temp %>%
+    left_join(Brood_data_sel,
+              by = c("BreedingSeason", "Species", "PopID", "LocationID", "BroodID")) %>%
+    #### The age of captured chicks in days since hatching
+    dplyr::mutate(ChickAge = if_else(Age_observed == 1,
+                                     as.integer(difftime(CaptureDate, HatchDate_observed)),
+                                     NA_integer_),
+                  BroodIDLaid = BroodID) %>%
+
+    #### NEW VERSION OF THE FUNCTION (TO FINISH!!)
+    # dplyr::mutate(Age_calculated = calc_age())
+
+    #### OLD VERSION OF THE FUNCTION
+    calc_age(ID = IndvID,
+             Age = Age_observed,
+             Date = CaptureDate,
+             Year = BreedingSeason,
+             showpb = TRUE) %>%
+    #### Final arrangement
+    dplyr::select(CaptureID, IndvID, Species, Sex_observed, BreedingSeason,
+                  CaptureDate, CaptureTime, ObserverID, LocationID,
+                  CaptureAlive, ReleaseAlive, CapturePopID, CapturePlot,
+                  ReleasePopID, ReleasePlot, Mass, Tarsus, OriginalTarsusMethod,
+                  WingLength, Age_observed, Age_calculated, ChickAge, ExperimentID)
+
+    # #### Remove records with no date
+    # #### CHECK: doing this, almost 500 records of chicks will be removed,
+    # #### as there is no CaptureDate !!!
```

Comment:
Many of the chick records do have a capture date, they just need to be calculated from the data. e.g. 13617071 has a capture record where the date is 'D14', which I assume is 14 days after hatching (need to ask the data owner to be sure), which would give it a capture date of 2016-05-28 (i.e. the hatch date of the brood 2016-05-14 + 14)

Reply:
Oh, I totally missed that "D14" information! I assume is that after the transformation of the column to date, all non-date characters were changed to NAs... 🤦‍♀️
Sure, now I can do the calculation from the hatching date. Thanks!